src/agent/communicator/src/http_request_params.cpp

```diff
@@ -23,14 +23,27 @@ namespace http_client
 
         if (!result)
         {
-            LogError("Invalid URL: {}", result.error().message());
+            LogError("Invalid URL: {}. Error: {}", serverUrl, result.error().message());
             return;
         }
 
         const auto& url = *result;
-        Use_Https = url.scheme().empty() || url.scheme() == "https";
+
+        if (url.scheme().empty() || (url.scheme() != "http" && url.scheme() != "https"))
+        {
+            LogError("Invalid URL scheme: {}", serverUrl);
+            return;
+        }
+
+        if (url.host().empty())
+        {
+            LogError("Invalid URL host: {}", serverUrl);
+            return;
+        }
+
+        Use_Https = url.scheme() == "https";
```

Comment:
Tiny improvement, no need to check if the scheme is empty if checking that it is either http or https.

```
if (url.scheme() != "http" && url.scheme() != "https")
```